Fixed ticket #534

bi4646nv opened the pull request:
This pull request fixes ticket #534 of the Avatar Project. In that ticket, these instructions are given:
- Make the currently active tab change color and style:
- Background of that specific tab to be green, with the letter in yellow.

This is what it looked like prior:
<img width="1195" height="827" alt="Screenshot 2025-11-14 165646" src="https://github.com/user-attachments/assets/d7422202-6dfa-4424-a888-15c635ecf886" />
where the active tab would show up as a lighter grey. Both the top and bottom tab are selected, even though only one tab is shown at a time. (This issue is not included on this ticket, but it is fixed in my implementation.)

This is what my implementation looks like for the ticket #534:
<img width="1197" height="829" alt="Screenshot 2025-11-14 173120" src="https://github.com/user-attachments/assets/4dd9b224-f334-432b-a736-3e8e8d6dd099" />
where the active tab has a background that is green, and the letter is in yellow. It no longer has both a top and bottom tab selected, but rather just highlights the one that is actually selected. As you can see, it works for the bottom tab as well:
<img width="1193" height="827" alt="image" src="https://github.com/user-attachments/assets/c4e15f52-6dbd-4002-88c7-6229ab38ed90" />


(Extra additions are made to the plotDevelopers .png files, since they are reloaded with the execution of GUI5.py. They can be ignored or excluded if needed.)

3C-SCSU reviewed (commented):
Ticket needs to go through review. It is preapproved tough. First need to confirm why deleted plotDevelopers.

Solving ticket: https://github.com/3C-SCSU/Avatar/issues/534

bi4646nv commented:
To clarify, I **did not delete** plotDevelopers, but rather it was **updated** when the GUI5.py program was ran. I have reverted my original commit and changed it so that updates to plotDevelopers are not made as a consequence of running GUI5.py, so that only the main.qml file is adjusted.

In the main.qml code, the code has been adjusted so that rather than showing highlights on the top and bottom tab, it just shows which one is currently selected. The currently selected tab now has a green background with yellow text, as seen here:
<img width="1197" height="829" alt="Screenshot 2025-11-14 173120" src="https://github.com/user-attachments/assets/ba4a5182-758e-4d13-88ea-55529896c7ab" />

Regarding this comment on this pull request:
<img width="875" height="911" alt="image" src="https://github.com/user-attachments/assets/bff52516-b99b-40f7-bfc8-3cb75b3ddf3f" />
I **did not adjust any code affecting this**, as this is what the tab looks like prior to me working on the code:
<img width="2146" height="992" alt="Screenshot 2025-11-14 185047" src="https://github.com/user-attachments/assets/fc8bbdf4-5166-4b15-b00b-46a29eed1dd0" />
As seen in the above screenshot, the active tabs are still grey, so this is the original, unedited GUI5.py. This may be a rendering or screenshot error on my side, but overall, the code affecting the color of this box was not adjusted by me whatsoever.

bi4646nv commented:
I should also note that the screenshot in the original pull request comment was not made to demonstrate anything in the contents of the File Shuffler tab but rather to demonstrate the green background and yellow text on the active bottom tab.
Original screenshot.
<img width="1193" height="827" alt="image" src="https://github.com/user-attachments/assets/6f930641-3be2-4655-a518-d8b9a3a2e460" />

3C-SCSU commented:
@bi4646nv  Okay. I got it. Probably when the ticket for Drone bug fixing was done the background color for the log has changed. No worries then.

Now we just need the additional reviews. Then we will be able to merge it.

3C-SCSU reviewed (changes requested):
@bi4646nv When you have a chance could you please fix the conflict:
<img width="992" height="198" alt="image" src="https://github.com/user-attachments/assets/99cc365e-9055-4445-990d-fedbc4551485" />

bi4646nv commented:
When I resolved the changes in main.qml, for some reason ManualNaoControl.qml was deleted, so I had to revert that change in a second commit (being the fifth so far). In the "Files changed" in this pull request, only main.qml is adjusted:
<img width="1609" height="535" alt="image" src="https://github.com/user-attachments/assets/42cd82cb-45bc-4aca-a706-7e34abf6e1ad" />

3C-SCSU commented:
@bi4646nv Hope your team get the conflict solved and resubmit it soon. Ticket is open again and up for grabs: first come, first serve. https://github.com/3C-SCSU/Avatar/issues/534
